newrelic/core/config.py

```diff
@@ -73,6 +73,23 @@ def emit(self, record):
 _logger.addHandler(_NullHandler())
 
 
+def _map_aws_account_id(s):
+    # The AWS account id must be a 12 digit number.
+    # See https://docs.aws.amazon.com/accounts/latest/reference/manage-acct-identifiers.html#awsaccountid.
+    if s and len(s) == 12:
+        try:
+            account_id = int(s)
+            return account_id
+        except ValueError:
+            pass
+    # Only log a warning if s is set.
+    if s:
+        _logger.warning(
+            "Improper configuration. cloud.aws.account_id = %s will be ignored because it is not a 12 digit number.", s
+        )
+    return None
```

Comment:
```suggestion
    if s and len(s) == 12 and s.isdigit():
        return int(s)

    # Only log a warning if s is set.
    if s:
        _logger.warning(
            f"Improper configuration. cloud.aws.account_id = {s} will be ignored because it is not a 12 digit number."
        )

    return None
```

Reply:
isdigit looks cool! It looks like it also returns true if there is a power digit like ² which would result in a ValueError still being raised by the int cast though.
```
>>> s="01234²34"
>>> s.isdigit()
True
>>> int(s)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '01234²34'
```

Reply:
@hmstepanek you can use `s.isdecimal()` instead!